server/lib/src/valueset/auditlogstring.rs

```diff
@@ -0,0 +1,164 @@
+#![allow(warnings)]
+
+use smolset::SmolSet;
+
+use crate::prelude::*;
+use crate::repl::cid::Cid;
+use crate::repl::proto::ReplAttrV1;
+use crate::schema::SchemaAttribute;
+use crate::valueset::{DbValueSetV2, ValueSet};
+
+type AuditLogStringType = (Cid, String);
+
+#[derive(Debug, Clone)]
+pub struct ValueSetAuditLogString {
+    set: SmolSet<[AuditLogStringType; 8]>,
+}
+
+impl ValueSetAuditLogString {
+    fn remove_oldest(&mut self) {
+        let oldest = self.set.iter().min().cloned();
+        if let Some(oldest_value) = oldest {
+            self.set.remove(&oldest_value);
+        }
+    }
+
+    pub fn new(s: AuditLogStringType) -> Box<Self> {
+        let mut set = SmolSet::new();
+        set.insert(s);
+        Box::new(ValueSetAuditLogString { set })
+    }
+
+    pub fn push(&mut self, s: AuditLogStringType) -> bool {
+        self.set.insert(s)
+    }
+
+    pub fn from_dbvs2(data: Vec<AuditLogStringType>) -> Result<ValueSet, OperationError> {
+        let set = data.into_iter().collect();
+        Ok(Box::new(ValueSetAuditLogString { set }))
+    }
+
+    pub fn from_repl_v1(data: &[AuditLogStringType]) -> Result<ValueSet, OperationError> {
+        let set = data.iter().map(|e| (e.0.clone(), e.1.clone())).collect();
+        Ok(Box::new(ValueSetAuditLogString { set }))
+    }
+
+    // We need to allow this, because rust doesn't allow us to impl FromIterator on foreign
+    // types, and str is foreign
+    #[allow(clippy::should_implement_trait)]
+    pub fn from_iter<T>(iter: T) -> Option<Box<Self>>
+    where
+        T: IntoIterator<Item = AuditLogStringType>,
+    {
+        let set = iter.into_iter().collect();
+        Some(Box::new(ValueSetAuditLogString { set }))
+    }
+}
+
+impl ValueSetT for ValueSetAuditLogString {
+    fn insert_checked(&mut self, value: Value) -> Result<bool, OperationError> {
+        match value {
+            Value::AuditLogString(c, s) => {
+                if self.set.len() >= 8 {
```

Comment:
should be > 8, not >=.

Reply:
wait but wouldn't this allow to have 9 elements in the set? Suppose the current size is 9, using your condition we would remove the 9th element to then add a new 9th right back, bringing the size back to 9. Whereas if we have 8 elements, we remove the 8th and we add a new one, the max size will still be 8

Reply:
Just tried with > instead of >= and it's not passing the history full test 
`thread 'plugins::namehistory::tests::name_purge_and_set_with_filled_history' panicked at 'assertion failed: !c.contains(&PartialValue::new_utf8s(&\"old_name1\")) &&\n    c.contains(&PartialValue::new_utf8s(&\"new_name\"))', server/lib/src/plugins/namehistory.rs:261:24`